aten/src/TH/THRandom.c

```diff
@@ -305,6 +305,78 @@ double THRandom_standard_gamma(THGenerator *_generator, double alpha) {
   }
 }
 
+// This is identical to THRandom_standard_gamma but also stores the
+// reparameterized gradient wrt alpha in grad_alpha.
+double THRandom_standard_gamma_with_grad(THGenerator *_generator, double alpha,
```

Comment:
You need to compute the grad during forward, because you'd need to replay all the control flow here otherwise, right?

Reply:
Correct. This is a complicated gradient due to the rejection sampler. To reproduce the computation, would need to store about 10x more state.